.github/workflows/flathub-pr-check.yml

```diff
@@ -0,0 +1,86 @@
+name: Build Flatpak CI
+
+on:
+  pull_request:
+    branches:
+      - master
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+concurrency:
+  # auto-cancel previous runs if the pull request is force-push updated
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  build-flatpak:
+    runs-on: ubuntu-latest
+    steps:
+    # PR/CI
+    - name: Checkout relevant repo
+      uses: actions/checkout@v4
+      with:
+        path: "xbmc"
+    # Flathub config
+    - name: Checkout Flathub config
+      uses: actions/checkout@v4
+      with:
+        repository: 'flathub/tv.kodi.Kodi'
+        submodules: true
+        ref: "beta"
+        path: "flathub"
+    - name: Install dependencies
+      run: |
+        sudo apt-get update -y
+        DEBIAN_FRONTEND=noninteractive sudo apt-get install --force-yes -y cmake git build-essential ccache lsb-release flatpak-builder flatpak
```

Comment:
You shouldn't install flatpak-builder here, but install it from org.flatpak.Builder then run it via flatpak. Otherwise we might run into intransparent errors/warnings.

Reply:
The [docs](https://docs.flatpak.org/en/latest/first-build.html) mention
> You also need to have installed flatpak-builder, which is usually available from the same repository as the flatpak package (e.g. use apt or dnf). You can also install it as a flatpak with flatpak install flathub org.flatpak.Builder.

which is why I went with the first option. I assume it doesn't change anything further down when going with `flatpak install`, so I can just add an additional step to do that and leave everything else as-is?

Reply:
Flatpak docs are not flathub docs. It's true that it will work for flatpak, but the flatpaked one is patched for flathub AFAIK.

Reply:
Got it, I'll change it and see how that goes. I'm currently doing some more testing and benchmarking with the caching to see how that improves the build times.

Reply:
You will need to call builder with 'flatpak run org.flatpak.Builder.... " after the change, see flathub docs.

Reply:
Is that something we should add to the Makefile? Maybe as an option to specify the builder to use in an environment variable, and default to "flatpak-builder" if none is specified?

Reply:
We can. Personally I'm not big on make files - I also think it's only used on personal system/dev case and not in the CI.